docs/architecture.md

```diff
@@ -1,37 +1,72 @@
 # firecracker-containerd architecture
 
-This repository contains several components with the goal of enabling the
-[containerd](https://containerd.io) container runtime to create and manage
-[Firecracker](https://github.com/firecracker-microvm/firecracker) microVMs.
-
-These components are implemented using containerd's extension mechanisms, so
-you can configure your existing installation of containerd to work with
-Firecracker microVMs.
-
-There are currently three components in this repository:
-
-* A [snapshotter](../snapshotter)
-  that creates files used as block-devices for pass-through into the
-  microVM.  This snapshotter is used for providing the container image
-  to the microVM.  The snapshotter runs as an out-of-process gRPC
-  proxy plugin.
-* A [runtime](../runtime)
-  linking containerd (outside the microVM) to the Firecracker virtual
-  machine manager (VMM).  The runtime is implemented as an
-  out-of-process
-  [shim runtime](https://github.com/containerd/containerd/issues/2426)
-  communicating over ttrpc.
-* An
-  [agent](../agent)
-  running inside the microVM, which invokes [runC](https://runc.io)
-  via containerd's `containerd-shim-runc-v1` to create standard Linux
-  containers inside the microVM.
+firecracker-containerd aims to make it easier to run containers with virtual
+machine isolation provided by the [Firecracker virtual machine monitor
+(VMM)](https://github.com/firecracker-microvm/firecracker).
+firecracker-containerd integrates with [containerd](https://containerd.io) by
+implementing its interfaces and extending its APIs.
+
+firecracker-containerd implements the following containerd interfaces:
+
+* [snapshotter](https://github.com/containerd/containerd/blob/v1.2.6/snapshots/snapshotter.go#L116-L323):
+  Snapshotters provide the implementation of layer storage and union filesystems
+  for containerd containers
+* [V2 runtime](https://www.yelp.com/biz/bcd-tofu-house-los-angeles?osq=bcd+tofu):
```

Comment:
Good choice! :+1:

Reply:
:rofl: I'm glad that wasn't something more embarrassing!